### PR TITLE
Load derived to redshift

### PR DESCRIPTION
**Pull request by mreid-moz**
Fetch metadata for credentials, then load the data into Redshift.

**Warning: This has not yet been fully tested!**

### CODE REVIEW

File: reports/derived_streams/run.sh [hunk 1]
```diff
@@ -33,7 +33,55 @@ fi
...
+TOKEN=$(jq -r '.Token' < aws_creds.json)
+
+# See http://docs.aws.amazon.com/redshift/latest/dg/copy-parameters-credentials.html
+CREDS="aws_access_key_id=${AWS_KEY};aws_secret_access_key=${AWS_SECRET_KEY};token=${TOKEN}"
+for t in main crash executive; do
+    NEW_TABLE="${t}_summary_${TARGET}"
+    echo "Copying data for $NEW_TABLE..."
+    $PQ -c "CREATE TABLE IF NOT EXISTS $NEW_TABLE (LIKE ${t}_summary including defaults);"
+    $PQ -c "COPY $NEW_TABLE FROM 's3://telemetry-private-analysis-2/derived_streams/data/${NEW_TABLE}' CREDENTIALS '$CREDS' ACCEPTANYDATE TRUNCATECOLUMNS ESCAPE;"
+    $PQ -c "GRANT SELECT ON $NEW_TABLE TO read_only;"
```
Comment (trink) [inline reports/derived_streams/run.sh:93]: I thought `read_write` doesn't have permissions to grant

File: reports/derived_streams/run.sh [hunk 2]
```diff
@@ -33,7 +33,55 @@ fi
...
+
+# Get the Telemetry data location
+BUCKET=$(jq -r '.["telemetry"].bucket' < sources.json)
+PREFIX=$(jq -r '.["telemetry"].prefix' < sources.json)
+
 # Run code:
-../heka/bin/heka-s3list -schema schema.json -bucket='net-mozaws-prod-us-west-2-pipeline-data' -bucket-prefix='telemetry-2' > list.txt
+../heka/bin/heka-s3list -schema schema.json -bucket="$BUCKET" -bucket-prefix="$PREFIX" > list.txt
 lua splitter.lua
 ../hindsight/bin/hindsight_cli hindsight.cfg 7
```
Comment (trink) [inline reports/derived_streams/run.sh:48]: If the return value is non-zero then don't load into Redshift

File: reports/derived_streams/run.sh (hunk 1 above)
Comment (mreid-moz) [inline reports/derived_streams/run.sh:93]: Turns out it just didn't like the "grant on all tables" thing.

**Comment by mreid-moz** [comment]
Ok, a test finally completed successfully (for 2015-08-30).

**Pull request merged**